Add initial file structure

joeylouise opened the pull request:
This PR sets up the initial file structure, including following folders and files:

- public
- src
- database_build
- tests
- eslint
- package.json

Relates #1

joeylouise committed:
Add database folder and files
Relates #1

joeylouise committed:
Add public folder, containing scripts folder, d3 folder, helpers folder, main.js, main.css and index.html
Relates #1

joeylouise committed:
Add tests folder and dummy files
Relates #1

joeylouise committed:
Add src subfolders - database, helpers, routes
Relates #1

joeylouise committed:
Initialise node project with package.json
Add eslint as a dev dependency
Relates #1

joeylouise committed:
Add initial d3 setup
Relates #3

joeylouise committed:
Add initial d3 setup
Relates #3

des-des commented on public/scripts/d3/margins.js:
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable */
+
+var width = 1000;
```
Width here has global scope

You need to decide how you are going to modularize your code.

Using an iife is the easiest option, happy to go into more detail if you like

ConchobarMacNessa replied:
Hi, we wanted to give width/heigh etc global scope as it's likely we'll need to refer to it later on in our d3 code – if we use an iife I don't think we'll have access to the variable, right?

des-des commented on public/main.css:
```diff
@@ -0,0 +1,24 @@
+  height: 640px;
+}
+
+svg {
```
You want to apply this to all svgs?

des-des commented on public/index.html:
```diff
@@ -0,0 +1,20 @@
+  <body>
+    <div id="desktop-background" >
+      <div id="d3-canvas">
+
```
Hmm, are you planning on using a canvas element?

A canvas is a html5 thing that I expect you will not use. The name is misleading

des-des commented on public/scripts/d3/margins.js:
```diff
@@ -0,0 +1,14 @@
+
+var svg = d3.select("#d3-canvas")
+  .append("svg")
+    // .attr("background-color", "pink")
```
No comments please

des-des replied:
I mean commented out code. If you want to merge some commented code into master, then leave a comment to explain why it needs to stay

des-des commented:
Final thing, try to add issue numbers to *all* commits, not sure I might have forgot to mention that

alice-carr committed:
updated PR as per des-des comments
placed global variables in iife in margin js
removed comment
specified css and removed svg
renamed d3 canvas element
relates #1 #3
